daemon/api.go

```diff
@@ -420,6 +405,72 @@ func shouldSearchStore(r *http.Request) bool {
 	return false
 }
 
+func storeUpdates(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	found, err := allLocalSnapInfos(c.d.overlord.State())
+	if err != nil {
+		return InternalError("cannot list local snaps: %v", err)
+	}
+
+	localSnapsInfo := make([]*store.CurrentSnap, len(found))
+	localSnapMap := map[string]*snap.Info{}
+	for i, sn := range found {
+		localSnapsInfo[i] = &store.CurrentSnap{
+			// the desired channel (not sn.info.Channel!)
+			Channel: sn.snapst.Channel,
+
+			SnapID:      sn.info.SnapID,
+			Revision:    sn.info.Revision,
+			Epoch:       sn.info.Epoch,
+			Confinement: sn.info.Confinement,
```

Comment:
we should also keep in mind that we don't want things in try to be refreshed except if they are manually requested to be. not super clear where that decision should happen though, it would still be good if snap refresh --list told the user what will happen if invoking snap refresh try-snap